src/app/views/app-sections/StatusMessages.tsx

```diff
@@ -4,31 +4,32 @@ import { FormattedMessage } from 'react-intl';
 
 import { IQuery } from '../../../types/query-runner';
 import { GRAPH_URL } from '../../services/graph-constants';
-import { convertArrayToObject, extractUrl, replaceLinks } from '../../utils/status-message';
+import {
+  convertArrayToObject, extractUrl, getMatchesAndParts,
+  matchIncludesLink, replaceLinks
+} from '../../utils/status-message';
 
 export function statusMessages(queryState: any, sampleQuery: IQuery, actions: any) {
   function displayStatusMessage(message: string, urls: any) {
-    const pattern = /([$0-9]+)/g;
-    message = message.toString();
-    const matches = message.match(pattern);
-    const parts = message.split(pattern);
+    const { matches, parts } = getMatchesAndParts(message);
 
     if (!parts || !matches || !urls || Object.keys(urls).length === 0) {
       return message;
     }
 
-
     return parts.map((part: string, index: number) => {
+      const includesLink = matchIncludesLink(matches, part);
       const displayLink = (): React.ReactNode => {
         const link = urls[part];
-        if (link.includes(GRAPH_URL)) {
-          return <Link onClick={() => setQuery(link)}>{link}</Link>;
+        if (link) {
```

Comment:
This will be cleaner if you use `&&`.

```js
if (link && link.includes(GRAPH_URL)) {
...
}
```

Reply:
@jobala I am not sure how I would handle the nested nature of the if statements?
Since 
- if there is no link, it should not return anything
- if there is a link it checks
-- should return a clickable graph link if the link is related to the graph URL
-- open a new tab if not related to the graph

Reply:
I see makes sense.